File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-com-5-4/OilDelivery.cpp
// OilDelivery.cpp 
#include "OilDelivery.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

OilDelivery::OilDelivery(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //pipelineNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* pipelineNode = xmlNode->FirstChildElement("pipeline");

   if(pipelineNode){pipelineIsNull_ = false;}
   else{pipelineIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- pipelineNode , address : " << pipelineNode << std::endl;
   #endif
   if(pipelineNode)
   {
      if(pipelineNode->Attribute("href") || pipelineNode->Attribute("id"))
      {
          if(pipelineNode->Attribute("id"))
          {
             pipelineIDRef_ = pipelineNode->Attribute("id");
             pipeline_ = boost::shared_ptr<OilPipelineDelivery>(new OilPipelineDelivery(pipelineNode));
             pipeline_->setID(pipelineIDRef_);
             IDManager::instance().setID(pipelineIDRef_,pipeline_);
          }
          else if(pipelineNode->Attribute("href")) { pipelineIDRef_ = pipelineNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { pipeline_ = boost::shared_ptr<OilPipelineDelivery>(new OilPipelineDelivery(pipelineNode));}
   }

   //transferNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* transferNode = xmlNode->FirstChildElement("transfer");

   if(transferNode){transferIsNull_ = false;}
   else{transferIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- transferNode , address : " << transferNode << std::endl;
   #endif
   if(transferNode)
   {
      if(transferNode->Attribute("href") || transferNode->Attribute("id"))
      {
          if(transferNode->Attribute("id"))
          {
             transferIDRef_ = transferNode->Attribute("id");
             transfer_ = boost::shared_ptr<OilTransferDelivery>(new OilTransferDelivery(transferNode));
             transfer_->setID(transferIDRef_);
             IDManager::instance().setID(transferIDRef_,transfer_);
          }
          else if(transferNode->Attribute("href")) { transferIDRef_ = transferNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { transfer_ = boost::shared_ptr<OilTransferDelivery>(new OilTransferDelivery(transferNode));}
   }

   //importerOfRecordNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* importerOfRecordNode = xmlNode->FirstChildElement("importerOfRecord");

   if(importerOfRecordNode){importerOfRecordIsNull_ = false;}
   else{importerOfRecordIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- importerOfRecordNode , address : " << importerOfRecordNode << std::endl;
   #endif
   if(importerOfRecordNode)
   {
      if(importerOfRecordNode->Attribute("href") || importerOfRecordNode->Attribute("id"))
      {
          if(importerOfRecordNode->Attribute("id"))
          {
             importerOfRecordIDRef_ = importerOfRecordNode->Attribute("id");
             importerOfRecord_ = boost::shared_ptr<PartyReference>(new PartyReference(importerOfRecordNode));
             importerOfRecord_->setID(importerOfRecordIDRef_);
             IDManager::instance().setID(importerOfRecordIDRef_,importerOfRecord_);
          }
          else if(importerOfRecordNode->Attribute("href")) { importerOfRecordIDRef_ = importerOfRecordNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { importerOfRecord_ = boost::shared_ptr<PartyReference>(new PartyReference(importerOfRecordNode));}
   }

   //absoluteToleranceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* absoluteToleranceNode = xmlNode->FirstChildElement("absoluteTolerance");

   if(absoluteToleranceNode){absoluteToleranceIsNull_ = false;}
   else{absoluteToleranceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- absoluteToleranceNode , address : " << absoluteToleranceNode << std::endl;
   #endif
   if(absoluteToleranceNode)
   {
      if(absoluteToleranceNode->Attribute("href") || absoluteToleranceNode->Attribute("id"))
      {
          if(absoluteToleranceNode->Attribute("id"))
          {
             absoluteToleranceIDRef_ = absoluteToleranceNode->Attribute("id");
             absoluteTolerance_ = boost::shared_ptr<AbsoluteTolerance>(new AbsoluteTolerance(absoluteToleranceNode));
             absoluteTolerance_->setID(absoluteToleranceIDRef_);
             IDManager::instance().setID(absoluteToleranceIDRef_,absoluteTolerance_);
          }
          else if(absoluteToleranceNode->Attribute("href")) { absoluteToleranceIDRef_ = absoluteToleranceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { absoluteTolerance_ = boost::shared_ptr<AbsoluteTolerance>(new AbsoluteTolerance(absoluteToleranceNode));}
   }

   //percentageToleranceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* percentageToleranceNode = xmlNode->FirstChildElement("percentageTolerance");

   if(percentageToleranceNode){percentageToleranceIsNull_ = false;}
   else{percentageToleranceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- percentageToleranceNode , address : " << percentageToleranceNode << std::endl;
   #endif
   if(percentageToleranceNode)
   {
      if(percentageToleranceNode->Attribute("href") || percentageToleranceNode->Attribute("id"))
      {
          if(percentageToleranceNode->Attribute("id"))
          {
             percentageToleranceIDRef_ = percentageToleranceNode->Attribute("id");
             percentageTolerance_ = boost::shared_ptr<PercentageTolerance>(new PercentageTolerance(percentageToleranceNode));
             percentageTolerance_->setID(percentageToleranceIDRef_);
             IDManager::instance().setID(percentageToleranceIDRef_,percentageTolerance_);
          }
          else if(percentageToleranceNode->Attribute("href")) { percentageToleranceIDRef_ = percentageToleranceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { percentageTolerance_ = boost::shared_ptr<PercentageTolerance>(new PercentageTolerance(percentageToleranceNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<OilPipelineDelivery> OilDelivery::getPipeline()
{
   if(!this->pipelineIsNull_){
        if(pipelineIDRef_ != ""){
             return boost::shared_static_cast<OilPipelineDelivery>(IDManager::instance().getID(pipelineIDRef_));
        }else{
             return this->pipeline_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<OilPipelineDelivery>();
   }
}
boost::shared_ptr<OilTransferDelivery> OilDelivery::getTransfer()
{
   if(!this->transferIsNull_){
        if(transferIDRef_ != ""){
             return boost::shared_static_cast<OilTransferDelivery>(IDManager::instance().getID(transferIDRef_));
        }else{
             return this->transfer_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<OilTransferDelivery>();
   }
}
boost::shared_ptr<PartyReference> OilDelivery::getImporterOfRecord()
{
   if(!this->importerOfRecordIsNull_){
        if(importerOfRecordIDRef_ != ""){
             return boost::shared_static_cast<PartyReference>(IDManager::instance().getID(importerOfRecordIDRef_));
        }else{
             return this->importerOfRecord_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PartyReference>();
   }
}
boost::shared_ptr<AbsoluteTolerance> OilDelivery::getAbsoluteTolerance()
{
   if(!this->absoluteToleranceIsNull_){
        if(absoluteToleranceIDRef_ != ""){
             return boost::shared_static_cast<AbsoluteTolerance>(IDManager::instance().getID(absoluteToleranceIDRef_));
        }else{
             return this->absoluteTolerance_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<AbsoluteTolerance>();
   }
}
boost::shared_ptr<PercentageTolerance> OilDelivery::getPercentageTolerance()
{
   if(!this->percentageToleranceIsNull_){
        if(percentageToleranceIDRef_ != ""){
             return boost::shared_static_cast<PercentageTolerance>(IDManager::instance().getID(percentageToleranceIDRef_));
        }else{
             return this->percentageTolerance_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PercentageTolerance>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-com-5-4/OilDelivery.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class OilDelivery : ISerialized
    {
        public OilDelivery(XmlNode xmlNode)
        {
            XmlNode pipelineNode = xmlNode.SelectSingleNode("pipeline");
            
            if (pipelineNode != null)
            {
                if (pipelineNode.Attributes["href"] != null || pipelineNode.Attributes["id"] != null) 
                {
                    if (pipelineNode.Attributes["id"] != null) 
                    {
                        pipelineIDRef_ = pipelineNode.Attributes["id"].Value;
                        OilPipelineDelivery ob = new OilPipelineDelivery(pipelineNode);
                        IDManager.SetID(pipelineIDRef_, ob);
                    }
                    else if (pipelineNode.Attributes["href"] != null)
                    {
                        pipelineIDRef_ = pipelineNode.Attributes["href"].Value;
                    }
                    else
                    {
                        pipeline_ = new OilPipelineDelivery(pipelineNode);
                    }
                }
                else
                {
                    pipeline_ = new OilPipelineDelivery(pipelineNode);
                }
            }
            
        
            XmlNode transferNode = xmlNode.SelectSingleNode("transfer");
            
            if (transferNode != null)
            {
                if (transferNode.Attributes["href"] != null || transferNode.Attributes["id"] != null) 
                {
                    if (transferNode.Attributes["id"] != null) 
                    {
                        transferIDRef_ = transferNode.Attributes["id"].Value;
                        OilTransferDelivery ob = new OilTransferDelivery(transferNode);
                        IDManager.SetID(transferIDRef_, ob);
                    }
                    else if (transferNode.Attributes["href"] != null)
                    {
                        transferIDRef_ = transferNode.Attributes["href"].Value;
                    }
                    else
                    {
                        transfer_ = new OilTransferDelivery(transferNode);
                    }
                }
                else
                {
                    transfer_ = new OilTransferDelivery(transferNode);
                }
            }
            
        
            XmlNode importerOfRecordNode = xmlNode.SelectSingleNode("importerOfRecord");
            
            if (importerOfRecordNode != null)
            {
                if (importerOfRecordNode.Attributes["href"] != null || importerOfRecordNode.Attributes["id"] != null) 
                {
                    if (importerOfRecordNode.Attributes["id"] != null) 
                    {
                        importerOfRecordIDRef_ = importerOfRecordNode.Attributes["id"].Value;
                        PartyReference ob = new PartyReference(importerOfRecordNode);
                        IDManager.SetID(importerOfRecordIDRef_, ob);
                    }
                    else if (importerOfRecordNode.Attributes["href"] != null)
                    {
                        importerOfRecordIDRef_ = importerOfRecordNode.Attributes["href"].Value;
                    }
                    else
                    {
                        importerOfRecord_ = new PartyReference(importerOfRecordNode);
                    }
                }
                else
                {
                    importerOfRecord_ = new PartyReference(importerOfRecordNode);
                }
            }
            
        
            XmlNode absoluteToleranceNode = xmlNode.SelectSingleNode("absoluteTolerance");
            
            if (absoluteToleranceNode != null)
            {
                if (absoluteToleranceNode.Attributes["href"] != null || absoluteToleranceNode.Attributes["id"] != null) 
                {
                    if (absoluteToleranceNode.Attributes["id"] != null) 
                    {
                        absoluteToleranceIDRef_ = absoluteToleranceNode.Attributes["id"].Value;
                        AbsoluteTolerance ob = new AbsoluteTolerance(absoluteToleranceNode);
                        IDManager.SetID(absoluteToleranceIDRef_, ob);
                    }
                    else if (absoluteToleranceNode.Attributes["href"] != null)
                    {
                        absoluteToleranceIDRef_ = absoluteToleranceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        absoluteTolerance_ = new AbsoluteTolerance(absoluteToleranceNode);
                    }
                }
                else
                {
                    absoluteTolerance_ = new AbsoluteTolerance(absoluteToleranceNode);
                }
            }
            
        
            XmlNode percentageToleranceNode = xmlNode.SelectSingleNode("percentageTolerance");
            
            if (percentageToleranceNode != null)
            {
                if (percentageToleranceNode.Attributes["href"] != null || percentageToleranceNode.Attributes["id"] != null) 
                {
                    if (percentageToleranceNode.Attributes["id"] != null) 
                    {
                        percentageToleranceIDRef_ = percentageToleranceNode.Attributes["id"].Value;
                        PercentageTolerance ob = new PercentageTolerance(percentageToleranceNode);
                        IDManager.SetID(percentageToleranceIDRef_, ob);
                    }
                    else if (percentageToleranceNode.Attributes["href"] != null)
                    {
                        percentageToleranceIDRef_ = percentageToleranceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        percentageTolerance_ = new PercentageTolerance(percentageToleranceNode);
                    }
                }
                else
                {
                    percentageTolerance_ = new PercentageTolerance(percentageToleranceNode);
                }
            }
            
        
        }
        
    
        #region Pipeline_
        private OilPipelineDelivery pipeline_;
        public OilPipelineDelivery Pipeline_
        {
            get
            {
                if (this.pipeline_ != null)
                {
                    return this.pipeline_; 
                }
                else if (this.pipelineIDRef_ != null)
                {
                    pipeline_ = IDManager.getID(pipelineIDRef_) as OilPipelineDelivery;
                    return this.pipeline_; 
                }
                else
                {
                      return this.pipeline_; 
                }
            }
            set
            {
                if (this.pipeline_ != value)
                {
                    this.pipeline_ = value;
                }
            }
        }
        #endregion
        
        public string pipelineIDRef_ { get; set; }
        #region Transfer_
        private OilTransferDelivery transfer_;
        public OilTransferDelivery Transfer_
        {
            get
            {
                if (this.transfer_ != null)
                {
                    return this.transfer_; 
                }
                else if (this.transferIDRef_ != null)
                {
                    transfer_ = IDManager.getID(transferIDRef_) as OilTransferDelivery;
                    return this.transfer_; 
                }
                else
                {
                      return this.transfer_; 
                }
            }
            set
            {
                if (this.transfer_ != value)
                {
                    this.transfer_ = value;
                }
            }
        }
        #endregion
        
        public string transferIDRef_ { get; set; }
        #region ImporterOfRecord_
        private PartyReference importerOfRecord_;
        public PartyReference ImporterOfRecord_
        {
            get
            {
                if (this.importerOfRecord_ != null)
                {
                    return this.importerOfRecord_; 
                }
                else if (this.importerOfRecordIDRef_ != null)
                {
                    importerOfRecord_ = IDManager.getID(importerOfRecordIDRef_) as PartyReference;
                    return this.importerOfRecord_; 
                }
                else
                {
                      return this.importerOfRecord_; 
                }
            }
            set
            {
                if (this.importerOfRecord_ != value)
                {
                    this.importerOfRecord_ = value;
                }
            }
        }
        #endregion
        
        public string importerOfRecordIDRef_ { get; set; }
        #region AbsoluteTolerance_
        private AbsoluteTolerance absoluteTolerance_;
        public AbsoluteTolerance AbsoluteTolerance_
        {
            get
            {
                if (this.absoluteTolerance_ != null)
                {
                    return this.absoluteTolerance_; 
                }
                else if (this.absoluteToleranceIDRef_ != null)
                {
                    absoluteTolerance_ = IDManager.getID(absoluteToleranceIDRef_) as AbsoluteTolerance;
                    return this.absoluteTolerance_; 
                }
                else
                {
                      return this.absoluteTolerance_; 
                }
            }
            set
            {
                if (this.absoluteTolerance_ != value)
                {
                    this.absoluteTolerance_ = value;
                }
            }
        }
        #endregion
        
        public string absoluteToleranceIDRef_ { get; set; }
        #region PercentageTolerance_
        private PercentageTolerance percentageTolerance_;
        public PercentageTolerance PercentageTolerance_
        {
            get
            {
                if (this.percentageTolerance_ != null)
                {
                    return this.percentageTolerance_; 
                }
                else if (this.percentageToleranceIDRef_ != null)
                {
                    percentageTolerance_ = IDManager.getID(percentageToleranceIDRef_) as PercentageTolerance;
                    return this.percentageTolerance_; 
                }
                else
                {
                      return this.percentageTolerance_; 
                }
            }
            set
            {
                if (this.percentageTolerance_ != value)
                {
                    this.percentageTolerance_ = value;
                }
            }
        }
        #endregion
        
        public string percentageToleranceIDRef_ { get; set; }
        public string choiceStr_0;
        public string choiceStr_1;
        
    
        
    
    }
    
}

